editor/plugins/node_3d_editor_plugin.cpp

```diff
@@ -1488,6 +1488,10 @@ Transform3D Node3DEditorViewport::_compute_transform(TransformMode p_mode, const
 }
 
 void Node3DEditorViewport::_surface_mouse_enter() {
+	if (Input::get_singleton()->get_mouse_mode() == Input::MOUSE_MODE_CAPTURED) {
```

Comment:
When the mouse cursor is captured by another element (i.e a `SpinSlider`), 3d viewport grabs focus hijacks input. It happens only if it's in the middle of the screen at that time, so must be unintentional

Reply:
I've had this rarely occur even outside of this PR. This sounds good to me on paper.

Reply:
This section is a duplicate of https://github.com/godotengine/godot/pull/87642, I can confirm that it works. All good if this PR supersedes it since it does more than mine.